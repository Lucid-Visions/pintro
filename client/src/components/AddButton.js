
import React from "react"
import { TouchableOpacity, Text, Image, View, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

class AddButton extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            image: './assets/addButton.png',
            action: props.action
        }
    }

    render() {
        return (
            <TouchableOpacity style={styles.btn} onPress={this.state.action}>
                <Ionicons name="ios-add-circle-outline" size={20} color="grey"/>
            </TouchableOpacity>
        )
    }
}

const styles = StyleSheet.create({
    btn: {
        alignItems: 'center',
        marginVertical: 10,
    }
});

export default AddButton