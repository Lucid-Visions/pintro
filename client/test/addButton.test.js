import React from 'react';
import renderer from 'react-test-renderer';
import AddButton from '../src/components/AddButton';

test('Button renders correctly', () => {
    const component = renderer.create(<AddButton />).toJSON();
    expect(component).toMatchSnapshot();
});

test('Action passed to button performed', () => {
    const actionFunction = () => console.log("hello");
    const component = renderer.create(<AddButton action={actionFunction} />);
    const instance = component.getInstance();
    expect(instance.state.action).toBe(actionFunction);
})

