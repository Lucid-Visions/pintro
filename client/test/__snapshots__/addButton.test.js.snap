// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "marginVertical": 10,
      "opacity": 1,
    }
  }
>
  <Text />
</View>
`;
